opus/application/static_media/js/opus.js

```diff
@@ -724,11 +734,91 @@ var opus = {
+        let userAgent = navigator.userAgent;
+        if (userAgent.indexOf("Firefox") > -1) {
+            // Example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:66.0)
+            // Gecko/20100101 Firefox/66.0
+            matchObj = userAgent.match(/Firefox\/(\d+.\d+)/);
+            browserName = "firefox";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Edge") > -1 || userAgent.indexOf("Edg") > -1) {
+            // userAgent example output:
+            // Windows: Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.10136
+            // Mac: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/76.0.3800.0 Safari/537.36 Edg/76.0.167.0
+            matchObj = userAgent.match(/Edge\/(\d+.\d+)/) || userAgent.match(/Edg\/(\d+.\d+)/);
+            browserName = "edge";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("OPR") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36 OPR/60.0.3255.95
+            matchObj = userAgent.match(/OPR\/(\d+.\d+)/);
+            browserName = "opera";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Chrome") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36
+            matchObj = userAgent.match(/Chrome\/(\d+.\d+)/);
+            browserName = "chrome";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Version") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15
+            // (KHTML, like Gecko) Version/12.1 Safari/605.1.15
+            matchObj = userAgent.match(/Version\/(\d+.\d+)/);
+            browserName = "safari";
+            browserVersion = matchObj[1];
+        }
+
+        if (opus.browserSupport[browserName] === undefined) {
+            $("#op-browser-version-msg").modal("show");
+        } else {
+            if (parseFloat(browserVersion) < opus.browserSupport[browserName]) {
+                $("#op-browser-version-msg").modal("show");
+            }
+        }
+    },
 
+    checkBrowserSize: function() {
+        /**
+         * Check if browser width is less than 1280px. If so, display a
+         * modal to inform the user to resize the browser width.
+         */
+        if ($(window).width() < opus.browserSupport.width) {
+            $("#op-browser-size-msg").modal("show");
+        } else {
+            $("#op-browser-size-msg").modal("hide");
+        }
+    },
+
+    checkCookies: function() {
```

Comment:
We're going to get rid of the widgets cookie, which doesn't currently do anything anyway. We should use some unique cookie for determining if the user is new.

Reply:
That means we should also set the cookie here. Don't rely on the widget code to do it for you.

Reply:
Now we check and set visited cookies here.